packages/backend-for-frontend/test/api/rateLimiterMiddleware.test.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi } from "vitest";
+import { generateId } from "pagopa-interop-models";
+import { generateToken } from "pagopa-interop-commons-test";
+import { authRole } from "pagopa-interop-commons";
+import request from "supertest";
+import { api, mockRateLimiter, services } from "../vitest.api.setup.js";
+import { appBasePath } from "../../src/config/appBasePath.js";
+import { getMockBffApiPurpose } from "../mockUtils.js";
+
+/*
+  Testing only the rateLimiterMiddleware, not the rate limiter itself,
+  which is tested in packages/commons-test/test/redisRateLimiter.integration.test.ts
+*/
+describe("rateLimiterMiddleware", () => {
+  const makeRequest = async (token: string) =>
+    request(api)
+      .get(`${appBasePath}/purposes/${generateId()}`)
+      .set("Authorization", `Bearer ${token}`)
+      .send();
+  // ^ using GET /purposes/:purposeId as a dummy endpoint to test the middleware
+
+  services.purposeService.getPurpose = vi
+    .fn()
+    .mockResolvedValue(getMockBffApiPurpose());
+
+  const mockRateLimitByOrganization = vi.fn();
+  mockRateLimiter.rateLimitByOrganization = mockRateLimitByOrganization;
+
+  it("Should correctly set rate limiter response headers when rate limit is not reached", async () => {
```

Comment:
maybe we could also check that `rateLimitByOrganization` is called with the correct organization id? not sure, because we're not usually checking that service methods are called with the expected parameters in these API tests, but we did so in the test for M2M m2mAuthDataValidationMiddleware

Reply:
Done!